tasks/eth/006-MCP-L1/postCheck.s.sol

```diff
@@ -0,0 +1,231 @@
+        require(resourceConfig.systemTxMaxGas == rconfig.systemTxMaxGas);
+        require(resourceConfig.minimumBaseFee == rconfig.minimumBaseFee);
+        require(resourceConfig.maximumBaseFee == rconfig.maximumBaseFee);
+        // Depends on start block being set to 0 in `initialize`
+        uint256 cfgStartBlock = cfg.systemConfigStartBlock();
+        require(config.startBlock() == (cfgStartBlock == 0 ? block.number : cfgStartBlock));
+        require(config.batchInbox() == cfg.batchInboxAddress());
+        // Check _addresses
+        require(config.l1CrossDomainMessenger() == prox.L1CrossDomainMessenger);
+        require(config.l1ERC721Bridge() == prox.L1ERC721Bridge);
+        require(config.l1StandardBridge() == prox.L1StandardBridge);
+        require(config.l2OutputOracle() == prox.L2OutputOracle);
+        require(config.optimismPortal() == prox.OptimismPortal);
+        require(config.optimismMintableERC20Factory() == prox.OptimismMintableERC20Factory);
+    }
+
+    /// @notice Asserts that the L1CrossDomainMessenger is setup correctly
+    function checkL1CrossDomainMessenger() internal view {
+        console.log("Running chain assertions on the L1CrossDomainMessenger");
+        L1CrossDomainMessenger messenger = L1CrossDomainMessenger(prox.L1CrossDomainMessenger);
+
+        require(address(messenger.OTHER_MESSENGER()) == Predeploys.L2_CROSS_DOMAIN_MESSENGER);
+        require(address(messenger.otherMessenger()) == Predeploys.L2_CROSS_DOMAIN_MESSENGER);
+
+        require(address(messenger.PORTAL()) == prox.OptimismPortal);
+        require(address(messenger.portal()) == prox.OptimismPortal);
+        require(address(messenger.superchainConfig()) == prox.SuperchainConfig);
+        bytes32 xdmSenderSlot = vm.load(address(messenger), bytes32(uint256(204)));
+        require(address(uint160(uint256(xdmSenderSlot))) == Constants.DEFAULT_L2_SENDER);
+    }
+
+    /// @notice Asserts that the L1StandardBridge is setup correctly
+    function checkL1StandardBridge() internal view {
+        console.log("Running chain assertions on the L1StandardBridge");
+        L1StandardBridge bridge = L1StandardBridge(payable(prox.L1StandardBridge));
+
+        require(address(bridge.MESSENGER()) == prox.L1CrossDomainMessenger);
+        require(address(bridge.messenger()) == prox.L1CrossDomainMessenger);
+        require(address(bridge.OTHER_BRIDGE()) == Predeploys.L2_STANDARD_BRIDGE);
+        require(address(bridge.otherBridge()) == Predeploys.L2_STANDARD_BRIDGE);
+        require(address(bridge.superchainConfig()) == prox.SuperchainConfig);
+    }
+
+    /// @notice Asserts that the L2OutputOracle is setup correctly
+    function checkL2OutputOracle() internal view {
+        console.log("Running chain assertions on the L2OutputOracle");
+        L2OutputOracle oracle = L2OutputOracle(prox.L2OutputOracle);
+
+        require(oracle.SUBMISSION_INTERVAL() == cfg.l2OutputOracleSubmissionInterval());
```

Comment:
Can we hardcode these values instead of reading them from the config? Then it'll remove misconfig issues

Reply:
I didn't fully understand -- should we hardcode all values from cfg? or just references to `cfg.l2OutputOracleSubmissionInterval()`?

Reply:
I think the suggestion here is that this script is for a specific task—the mainnet MCP upgrade—as a result we know precisely what values should be and therefore can hardcode expected values. For example, we do not want to change the submission interval as part of MCP, so let's hardcode the current value so we assert it was not changed

Reply:
I believe that no changes were made to `DeployConfig` as part of the upgrade. does this mean we can hardcode all values from `DeployConfig`? if so, what's the point of still importing the contract? we could skip it altogether it seems?

Reply:
thoughts @tynes @mds1?

Reply:
Hardcoding and removing the dependency on DeployConfig seems reasonable to me. I'd recommend hardcoding them by defining constants, like `uint256 constant expectedSubmissionInterval = 1800`